File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    nums.sort();/**오름차순 정렬  */
    /**Change the nums array to object that checks the num of elements */
    let temp = nums.reduce((accu,curr)=>{
        accu[curr] = (accu[curr]||0)+1;
        return accu;
    },{});
    /**Sort the object by descending order of value  */
    let downSorted = Object.entries(temp).sort((a,b)=>(b[1]-a[1]));
    let result = []
    /**Create the array for just */
    for(let element of downSorted){
        result.push(element[0]);
    }
    /**Cut off the key of the object*/
    result = result.slice(0,k); /**인덱스 0번째부터 k번째까지 자르기 */
    console.log(result);
    return result;
    
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """

        #Create dictionary which have a key(element of nums) and value(count num of the element)
        count = {}
        for i in nums:
            try: count[i] += 1
            except: count[i] = 1
        #print(count)
        
        #Sort depends on the value descending order
        sorted_count = sorted(count.items(),key=lambda x:x[1],reverse=True)
        dic_count = dict(sorted_count[:k])
        return dic_count.keys()

        #sorted_count = sorted(count.items(),key=lambda item:item[1],reverse=True)

        #0에서 k-1까지 key값을 반환 
        # answer = []
        # for i in range(k):
        #     answer.append(sorted_count.keys()[i])
        
        # return answer

        